Cargo.toml

```diff
@@ -41,6 +41,7 @@ async-sse = "4.0.1"
 async-std = { version = "1.6.5", features = ["unstable"] }
 async-trait = "0.1.41"
 femme = { version = "2.1.1", optional = true }
+futures = "0.3.7"
```

Comment:
We probably don't want to depend on `futures` directly since it adds about a minute of compile time in certain setups. Instead it's preferable to use smaller crates and/or async-std directly.

Reply:
Futures is required for "select": https://rust-lang.github.io/async-book/06_multiple_futures/03_select.html

Honestly, I don't see a way to do this without "select," otherwise you end up waiting for a final incoming http request before ending the loop accepting incoming requests. (See my comment about when I reviewed stop-token.)

Maybe it's worth trying to implement something like select manually, but, IMO, that might be a fools' errand.

Reply:
I think what we would prefer is [`futures_lite::future::race`](https://docs.rs/futures-lite/1.11.2/futures_lite/future/fn.race.html).

Reply:
Great advice!

Let me see what I can do. It'll probably be a few days before I can update this PR.